sdk/src/main/java/com/bugsnag/android/Bugsnag.java

```diff
@@ -75,11 +94,23 @@ public static Client init(@NonNull Context androidContext,
      */
     @NonNull
     public static Client init(@NonNull Context androidContext, @NonNull Configuration config) {
-        client = new Client(androidContext, config);
-        NativeInterface.configureClientObservers(client);
+        synchronized (lock) {
```

Comment:
Why is this code duplicated in all the init methods rather than calling one implementation init method from all the other ones?

Reply:
I considered putting the `Client` initialisation into a `Callable` (or equivalent) and passing that into a common method, but felt that this was outside the scope of changes originally agreed during design review. If you're happy with that approach I can alter the PR.

Reply:
I was just thinking in the other init methods creating a configuration object and then calling down to the other init function with that config object?

Reply:
I've updated the `init` methods so that they now do this.